src/extensions/manifest.ts

```diff
@@ -31,24 +34,44 @@ export async function writeToManifest(
     const currentExtensions = Object.entries(config.get("extensions"))
       .map((i) => `${i[0]}: ${i[1]}`)
       .join("\n\t");
-    const overwrite = await promptOnce({
-      type: "list",
-      message: `firebase.json already contains extensions:\n${currentExtensions}\nWould you like to overwrite or merge?`,
-      choices: [
-        { name: "Overwrite", value: true },
-        { name: "Merge", value: false },
-      ],
-    });
-    if (overwrite) {
-      config.set("extensions", {});
+    if (allowOverwrite) {
+      const overwrite = await promptOnce({
+        type: "list",
+        message: `firebase.json already contains extensions:\n${currentExtensions}\nWould you like to overwrite or merge?`,
+        choices: [
+          { name: "Overwrite", value: true },
+          { name: "Merge", value: false },
+        ],
+      });
+      if (overwrite) {
+        config.set("extensions", {});
+      }
     }
   }
 
   writeExtensionsToFirebaseJson(specs, config);
   await writeEnvFiles(specs, config, options.force);
 }
 
-// TODO(lihes): Add some tests.
+export function loadConfig(options: any): Config {
+  const existingConfig = Config.load(options, true);
+  if (!existingConfig) {
+    throw new FirebaseError(
+      "Not currently in a Firebase directory. Please run `firebase init` to create a Firebase directory."
+    );
+  }
+  return existingConfig;
+}
+
+/**
+ * Checks if an instance name already exists in the manifest.
+ */
+export function instanceExists(instanceId: string, config: Config): boolean {
+  return Object.keys(config.get("extensions", {})).some(
```

Comment:
Nit - realizing now that we can simplify this even further to: 
return !!config.get("extensions",{})[instanceId];